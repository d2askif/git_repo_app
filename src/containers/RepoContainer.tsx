import React, {Component} from 'react';
import {connect} from 'react-redux';
import {View, Text, SafeAreaView, StyleSheet} from 'react-native';
import {fetchRepos, starRepo} from '../redux/actions';
import List from '../components/list/list';
import ListItem from '../components/list/listItem';
import {State, Repo} from '../redux/store/types';

const styles = StyleSheet.create({
  safeAre: {
    flex: 1,
  },
  container: {
    flex: 1,
  },
});

interface Props {
  fetchRepos: () => void;
  toggleRepoStar: (id: number) => void;
  repos: any;
}

export class RepoContainer extends Component<Props> {
  constructor(props: Props) {
    super(props);
    this.renderItem = this.renderItem.bind(this);
    this.handleStarRepo = this.handleStarRepo.bind(this);
  }
  componentDidMount() {
    this.props.fetchRepos();
  }
  handleStarRepo(repoId: number) {
    const {toggleRepoStar} = this.props;
    toggleRepoStar(repoId);
  }

  renderItem(repo: Repo, index: number) {
    console.log('item', repo, 'index', index);
    return (
      <ListItem
        isStared={repo.stared}
        onStarPressed={() => this.handleStarRepo(repo.id)}
        name={repo.name}
        stargazers_count={repo.stargazers_count}
        avatar_url={repo.avatar_url}
        description={repo.description}
      />
    );
  }

  render() {
    return (
      // eslint-disable-next-line react-native/no-inline-styles
      <SafeAreaView style={styles.safeAre}>
        <View style={styles.container}>
          <Text>Test</Text>
          <List data={this.props.repos} renderItem={this.renderItem} />
        </View>
      </SafeAreaView>
    );
  }
}

const mapStateToProps = (state: State) => ({
  repos: state.repos,
});

const mapDispatchToProps = {
  fetchRepos,
  toggleRepoStar: starRepo,
};

// eslint-disable-next-line prettier/prettier
export default connect(mapStateToProps, mapDispatchToProps)(RepoContainer);
