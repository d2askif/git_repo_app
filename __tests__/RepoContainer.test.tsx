import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import RepoContainer from '../src/containers/RepoDetail';
test('RepoContainer snapshot', () => {
  const createTestProps = (props: Object) => ({
    navigation: {
      navigate: jest.fn(),
    },
    ...props,
  });
  const navigation = createTestProps({});
  const snapshot = renderer.create(
    <RepoContainer route={{params: {repo: {}}}} navigation={navigation} />,
  );
  expect(snapshot).toMatchSnapshot();
});
