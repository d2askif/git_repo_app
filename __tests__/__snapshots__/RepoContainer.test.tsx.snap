// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoContainer snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#CDE1F6",
        "height": 240,
      }
    }
  >
    <Text>
       
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#CDE1F6",
        "borderRadius": 90,
        "height": 240,
        "marginLeft": -40,
        "marginTop": -140,
        "width": 240,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 32,
        "bottom": 64,
        "elevation": 5,
        "left": 32,
        "position": "absolute",
        "right": 32,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
        "top": 120,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "height": 160,
          "marginTop": -76,
          "width": 160,
        }
      }
    >
      <View
        activeOpacity={0.7}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "white",
            "borderRadius": 75,
            "borderWidth": 4,
            "elevation": 2,
            "height": 150,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.15,
            "shadowRadius": 1.84,
            "width": 150,
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "#bdbdbd",
              "borderRadius": 75,
              "flex": 1,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            onLoad={[Function]}
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": null,
                  "width": null,
                },
              ]
            }
            testID="RNE__Image"
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          />
          <View
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "flex": 1,
                  "height": null,
                  "justifyContent": "center",
                  "width": null,
                }
              }
              testID="RNE__Image__placeholder"
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#ffffff",
                    "fontSize": 75,
                    "textAlign": "center",
                  }
                }
              >
                d
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "height": null,
                "width": null,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#999BB1",
            "fontSize": 24,
            "fontWeight": "300",
            "marginBottom": 16,
            "opacity": 1,
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 16,
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={Object {}}
        >
          <Text
            style={
              Object {
                "color": "#6D7195",
                "fontSize": 20,
                "opacity": 0.8,
              }
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#D7BFB6",
                    "fontSize": 15,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#999BB1",
                  "marginLeft": 4,
                }
              }
            >
              Fork
            </Text>
          </View>
        </View>
        <View
          style={Object {}}
        >
          <Text
            style={
              Object {
                "color": "#6D7195",
                "fontSize": 20,
                "opacity": 0.8,
              }
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#D7BFB6",
                    "fontSize": 15,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#999BB1",
                  "marginLeft": 4,
                }
              }
            >
              Watch
            </Text>
          </View>
        </View>
        <View
          style={Object {}}
        >
          <Text
            style={
              Object {
                "color": "#6D7195",
                "fontSize": 20,
                "opacity": 0.8,
              }
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#D7BFB6",
                    "fontSize": 15,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#999BB1",
                  "marginLeft": 4,
                }
              }
            >
              Star
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 16,
          "paddingHorizontal": 32,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6D7195",
            "fontSize": 14,
            "marginBottom": 8,
          }
        }
      >
        Description
      </Text>
      <Text
        style={
          Object {
            "color": "#999BB1",
            "fontSize": 14,
            "fontWeight": "300",
            "textAlign": "justify",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingBottom": 32,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 32,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2FA6F2",
                "borderColor": "#2089dc",
                "borderRadius": 32,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
                "width": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Share repo
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
